### PR TITLE
[MU4] Fix #334025: Export missing instrument-sound tag

### PR DESCRIPTION
**Pull request by Jojo-Schmitz**
Resolves: *https://musescore.org/en/node/334025*

Submitting this PR on behalf of @luto65

### CODE REVIEW

**Comment by RobFog** [comment]
Is this something for @lvinken to look at?

**Comment by Jojo-Schmitz** [comment]
It is based on his proposal at least ;-)

**Comment by Jojo-Schmitz** [comment]
Anyone knows a way to batch-fix all these (Music)XML fiiles that otherwise fail the unit tests?

**Comment by cbjeukendrup** [comment]
There used to be an `updateReferences` script somewhere in the repo... but I can't find it, and if we would find it, it's probably dramatically outdated so won't be useful.

If you can get the tests to run locally, you can try to find out where the files go that are produced by the tests, and copy them from there to the ref files in the repo. But I don't know from memory where to look.

**Comment by Jojo-Schmitz** [comment]
That `updateReferences` is still there, but just copies files, not regenerates them

**Comment by Jojo-Schmitz** [comment]
OK, for now I just circumvent the problem and disable that new functionaliy in test mode ;-)

**Comment by lvinken** [comment]
Comments from my side: I am aware of this change and agree with it. I would recommend against simply updating all testfiles. Almost all of them do not contain an instrument-sound element, which means they is generated on import and that needs to be checked for correctness. In 3.x there was indeed an updateReference script in mtest/musicxml/io, but I am not aware of its origin.

**Comment by Jojo-Schmitz** [comment]
OK, so I'll leave those disabled in test mode. The failing tests can get checked on in the 1st commit, it is all about that one line with `instrument-sound` missing, in 178 tests.

**Comment by Jojo-Schmitz** [comment]
Can this PR please get reviewd and merged?

File: src/importexport/musicxml/internal/musicxml/exportxml.cpp [hunk 1]
```diff
@@ -6525,7 +6529,9 @@ static void partList(XmlWriter& xml, Score* score, MxmlInstrumentMap& instrMap)
             MxmlReverseInstrumentMap rim;
             initReverseInstrMap(rim, instrMap);
             for (int instNr : rim.keys()) {
-                scoreInstrument(xml, static_cast<int>(idx) + 1, instNr + 1, MScoreTextToMXML::toPlainText(rim.value(instNr)->trackName()));
+                scoreInstrument(xml, static_cast<int>(idx) + 1, instNr + 1, MScoreTextToMXML::toPlainText(rim.value(
+                                                                                                              instNr)->trackName()), rim.value(
+                                    instNr));
```
Comment (cbjeukendrup) [inline src/importexport/musicxml/internal/musicxml/exportxml.cpp:6534]: Uncrustify has made this line quite unreadably. Maybe we can help it a bit by manually formatting it?
```suggestion
                scoreInstrument(xml, static_cast<int>(idx) + 1, instNr + 1, 
                                MScoreTextToMXML::toPlainText(rim.value(instNr)->trackName()), 
                                rim.value(instNr));
```

Comment (Jojo-Schmitz) [reply src/importexport/musicxml/internal/musicxml/exportxml.cpp:6534]: Done

File: src/importexport/musicxml/internal/musicxml/exportxml.cpp [hunk 2]
```diff
@@ -6017,10 +6017,14 @@ static int findPartGroupNumber(int* partGroupEnd)
...
 //---------------------------------------------------------
 
-static void scoreInstrument(XmlWriter& xml, const int partNr, const int instrNr, const QString& instrName)
+static void scoreInstrument(XmlWriter& xml, const int partNr, const int instrNr, const QString& instrName,
+                            const Instrument* instr = nullptr)
 {
     xml.startElementRaw(QString("score-instrument %1").arg(instrId(partNr, instrNr)));
     xml.tag("instrument-name", instrName);
+    if (instr && !instr->instrumentId().isEmpty() && !MScore::testMode) {
+        xml.tag("instrument-sound", instr->instrumentId());
```
Comment (cbjeukendrup) [inline src/importexport/musicxml/internal/musicxml/exportxml.cpp:6026]: Sorry, one more thing: `instrumentId()` has been renamed to `musicXmlId()` in the master branch.

Comment (Jojo-Schmitz) [reply src/importexport/musicxml/internal/musicxml/exportxml.cpp:6027]: Oops, needs to wait until tomorrow then

Comment (Jojo-Schmitz) [reply src/importexport/musicxml/internal/musicxml/exportxml.cpp]: Done (hopefully)

**Review by cbjeukendrup — APPROVED** [review]

**Comment by Jojo-Schmitz** [comment]
@RomanPudashkin mind to review and merge?

**Comment by Jojo-Schmitz** [comment]
@RomanPudashkin mind to review and merge?

**Pull request merged**